### DISCUSSION TITLE
Install extensions in in code launched through @vscode/test-electron

### DISCUSSION THREAD

**Discussion by ferenj-gs** [microsoft/vscode-discussions#1079] [Extension Development QnA]
Hey,

I want to write unit tests for my extension that is depending on cmake-tools. In my runTest.ts I try to install the extension dependencies with:
```typescript
        if (packageJson.extensionDependencies) {
            for (const extensionId of packageJson.extensionDependencies) {
                console.log(`Installing ${extensionId}`);
                cp.spawnSync(cli, [...args, '--force', '--install-extension', extensionId], {
                    encoding: 'utf-8',
                    stdio: 'inherit',
                });
            }
        }
```
When I launch it, I see that extensions-dir argument is not processed:
```bash
Ignoring option 'extensions-dir': not supported for code.
Ignoring option 'user-data-dir': not supported for code.
```
Also my tests are not working as my extension  activation fails, because activation fails, as in the started instance there are no extensions installed.

What is the recommended way for solving this issue? (i.e. extension has dependency, and the dependencies should be installed to this test-electron downloaded code instance.)

Thanks & BR,
Ferenc

**Comment by isidorn** [comment]
Not sure what's the best way forward here.
Are you still facing this challange? Have you made any progress? Thanks!

**Comment by abrilurena** [comment]
Hi @ferenj-gs! hope you're doing great. It's been a while since last time, do you still need help with this? thank you!

**Comment by abrilurena** [comment]
Hi, @ferenj-gs! it's been a while since last time, so i'm cosing this issue for now. Please feel free to re-open if you need more help ✨
